source/containers.tex

```diff
@@ -15032,3 +15034,2422 @@
+\pnum
+\effects
+\begin{itemize}
+\item
+Direct-non-list-initializes \exposid{ptr_} with \tcode{std::move(p)},
+\item
+direct-non-list-initializes \exposid{map_} with \tcode{m}, and
+\item
+direct-non-list-initializes \exposid{acc_} with \tcode{a}.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibraryctor{mdspan}%
+\begin{itemdecl}
+template<class OtherElementType, class OtherExtents,
+         class OtherLayoutPolicy, class OtherAccessor>
+  constexpr explicit(@\seebelow@)
+    mdspan(const mdspan<OtherElementType, OtherExtents,
+                        OtherLayoutPolicy, OtherAccessor>& other);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{is_constructible_v<mapping_type, const OtherLayoutPolicy::template mapping<Oth-\newline erExtents>\&>}
+is \tcode{true}, and
+\item
+\tcode{is_constructible_v<accessor_type, const OtherAccessor\&>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\mandates
+\begin{itemize}
+\item
+\tcode{is_constructible_v<pointer, const OtherAccessor::pointer\&>} is \tcode{true}, and
+\item
+\tcode{is_constructible_v<extents_type, OtherExtents>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\expects
+\begin{itemize}
+\item
+For each rank index \tcode{r} of \tcode{extents_type},
+\tcode{static_extent(r) == dynamic_extent || static_extent(r) == other.extent(r)}
+is \tcode{true}.
+\item
+$[0, \tcode{\exposid{map_}.required_span_size()})$ is
+an accessible range of \exposid{ptr_} and \exposid{acc_}
+for values of \exposid{ptr_}, \exposid{map_}, and \exposid{acc_}
+after the invocation of this constructor.
+\end{itemize}
+
+\pnum
+\effects
+\begin{itemize}
+\item
+Direct-non-list-initializes \exposid{ptr_} with \tcode{other.\exposid{ptr_}},
+\item
+direct-non-list-initializes \exposid{map_} with \tcode{other.\exposid{map_}}, and
+\item
+direct-non-list-initializes \exposid{acc_} with \tcode{other.\exposid{acc_}}.
+\end{itemize}
+
+\pnum
+\remarks
+The expression inside \tcode{explicit} is:
+\begin{codeblock}
+!is_convertible_v<const OtherLayoutPolicy::template mapping<OtherExtents>&, mapping_type>
+|| !is_convertible_v<const OtherAccessor&, accessor_type>
+\end{codeblock}
+\end{itemdescr}
+
+\rSec3[mdspan.mdspan.members]{Members}
+
+\indexlibrarymember{operator[]}{mdspan}%
+\begin{itemdecl}
+template<class... OtherSizeTypes>
+  constexpr reference operator[](OtherSizeTypes... indices) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{(is_convertible_v<OtherSizeTypes, size_type> \&\& ...)} is \tcode{true},
+\item
+\tcode{(is_nothrow_constructible_v<size_type, OtherSizeTypes> \&\& ...)} is \tcode{true}, and
+\item
+\tcode{sizeof...(OtherSizeTypes) == rank()} is \tcode{true}.
+\end{itemize}
+
+\pnum
+Let \tcode{I} be \tcode{extents_type::\exposid{index-cast}(std::move(indices))}.
+
+\pnum
+\expects
+\tcode{I} is a multidimensional index in \tcode{extents()}.
+\begin{note}
+This implies that
+\tcode{\exposid{map_}(I...) < \exposid{map_}.required_span_size()}
```

Comment:
```suggestion
\tcode{\exposid{map_}(I) < \exposid{map_}.required_span_size()}
```
Is this right? `I` is not a pack.

Reply:
This is what the incoming paper says.

@jwakely?

Reply:
[P0009R15](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p0009r15.html) said:

_Preconditions:_ `I...` is a multidimensional index into `extents()`. _[Note:_ This implies that `map_(I...) <map_.required_span_size()` is `true`.— _end note]_;

[P0009r16](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p0009r16.html) removed the ellipsis from the first `I...` but it looks like it should have been done in the note as well. I think it's OK to change this editorially.